packages/strapi-design-system/src/Select/Option.tsx

```diff
@@ -1,107 +1,66 @@
-    indeterminate &&
-    `&::after {
-      content: '';
-      display: block;
-      position: relative;
-      color: white;
-      height: 2px;
-      width: 10px;
-      background-color: ${theme.colors.neutral0};
-      left: 50%;
-      top: 50%;
-      transform: translateX(-50%) translateY(-50%);
-    }
-
-  `}
+import { SingleSelectOption, SingleSelectOptionProps } from './SingleSelect';
 
-  ${({ selected }) =>
-    selected &&
-    `  
-    &::after {
-      content: '';
-      background: url(${checkmarkIcon}) no-repeat no-repeat center center;
-      width: 100%;
-      height: 100%;
-      position: absolute;
-    }
-
-  `}
-`;
-
-const OptionBox = styled(Box)`
-  width: 100%;
-  border: none;
-  text-align: left;
-  outline-offset: -3px;
-  &.is-focused {
-    background: ${({ theme }) => theme.colors.primary100};
-  }
-
-  &:hover {
-    background: ${({ theme }) => theme.colors.primary100};
-  }
-`;
+import checkmarkIcon from '../BaseCheckbox/assets/checkmark.svg';
 
-export const Option = ({ selected, indeterminate, children, value, multi, isChild, startIcon, ...props }) => {
-  const optionRef = useRef(null);
+interface OptionProps extends SingleSelectOptionProps {
+  indeterminate?: boolean;
+  multi?: boolean;
+  isChild?: boolean;
+  selected?: boolean;
+}
 
-  return (
+export const Option = ({
+  indeterminate,
+  multi,
+  children,
+  value,
+  selected,
+  isChild,
+  startIcon,
+  ...restProps
+}: OptionProps) => {
+  return multi ? (
     <OptionBox
       as="li"
       hasRadius
       paddingLeft={isChild ? 7 : 4}
       paddingRight={4}
       paddingTop={2}
       paddingBottom={2}
-      ref={optionRef}
       role="option"
       aria-selected={selected}
       background="neutral0"
       data-strapi-value={value}
-      {...props}
+      {...restProps}
     >
       <Flex>
         {startIcon && (
           <Box paddingRight={2} aria-hidden>
             {startIcon}
           </Box>
         )}
-
-        {multi && (
-          <Box paddingRight={2} aria-hidden>
-            <CheckMark
-              hasRadius
-              overflow="hidden"
-              position="relative"
-              indeterminate={indeterminate}
-              selected={selected}
-              zIndex={1}
-            />
-          </Box>
-        )}
-        <Typography textColor={selected ? 'primary600' : 'neutral800'} fontWeight={selected ? 'bold' : null}>
+        <Box paddingRight={2} aria-hidden>
+          <CheckMark
+            hasRadius
+            overflow="hidden"
+            position="relative"
+            indeterminate={indeterminate}
+            selected={selected}
+            zIndex={1}
+          />
+        </Box>
+        <Typography textColor={selected ? 'primary600' : 'neutral800'} fontWeight={selected ? 'bold' : undefined}>
           {children}
         </Typography>
       </Flex>
     </OptionBox>
+  ) : (
+    <SingleSelectOption {...{ children, value, selected, isChild, startIcon }} />
```

Comment:
Does that mean the other branch is a `MultiSelectOption`? Would it make sense to improve readability to move it into it's own (local) component too?

Reply:
Its the multi select yep, I thought about it but didn't want to bloat this PR with that change, the next ticket addresses combobox which is what a multi-select is 😄 so then it'll be moved out – wdyt? still needs moving?

Reply:
No all good then 👍🏼